Lib/test/test_asyncio/test_tasks.py

```diff
@@ -2688,6 +2688,32 @@ def test_get_context(self):
         finally:
             loop.close()
 
+    def test_proper_refcounts(self):
+        # see: https://github.com/python/cpython/issues/126083
+        class Break:
+            def __str__(self):
+                raise RuntimeError("break")
+
+        obj = object()
+        initial_refcount = sys.getrefcount(obj)
+
+        coro = coroutine_function()
+        loop = asyncio.new_event_loop()
+        task = asyncio.Task.__new__(asyncio.Task)
+
+        for _ in range(5):
+            try:
+                task.__init__(coro, loop=loop, context=obj, name=Break())
+            except Exception as e:
```

Comment:
So update it to be this?
```python
for _ in range(5):
    with self.assertRaisesRegex(RuntimeError, 'break'):
        task.__init__(coro, loop=loop, context=obj, name=Break())
```

Reply:
FTR: In order to catch the correct exception in https://github.com/python/cpython/pull/126120, I used a custom exception + a special argument so that I'm sure that I catch *and* expect the correct one (and not just a random one because of something else). So what I did was something like:

```py
# what I want to catch
def foo():
	raise ReachableCode(1)

# what I don't want to catch yet
def bar():
	raise ReachableCode(2)

# where I call the thing I want to catch
with self.assertRaises(ReachableCode) as cm:
	something_that_calls_foo()
self.assertEqual(len(cm.exception.args), 1)
self.assertIs(cm.exception.args[0], 1)
```

I find this pattern quite good (maybe a bit too exhaustive) because I can really see what I'm catching.

Reply:
I like the idea of the ReachableCode exception class, but I still think it's easier and more readable to just do
```python
with self.assertRaisesRegex(ReachableCode, 'str here'):
    ...
```
and use a string in the arg instead of an int

Reply:
I think I'll go with that as well to simplify the tests on my side (but not today).